orttraining/orttraining/test/training_api/common/synthetic_data_loader.h

```diff
@@ -74,24 +57,29 @@ struct SyntheticSampleBatch {
   void AddInt32Input(const std::vector<int64_t>& shape, int32_t low, int32_t high);
```

Comment:
>onst std::vector<int64_t>& shap [](http://example.com/codeflow?start=22&length=31)

gsl::span<const int64_t> would enale users to pass many other kind of containers automatically.

Reply:
gsl::span and TensorShapeVector is not exposed as ORT C/C++ APIs, right? I did not use them and also Inlined versions because this file is not necessarily built together with ORT binaries, it can be build into a separated test_trainer exe, which simulates user's scripts leveraging our C/C++ training apis.